pages/installation/docker.rst

```diff
@@ -5,181 +5,284 @@ Docker
+log in with username ``admin`` and password ``admin`` (make sure to change the password later).
 
 How to get log data in
 ----------------------
 
 You can create different kinds of inputs under *System -> Inputs*, however you can only use ports that have been properly
 mapped to your docker container, otherwise data will not go through.
 
-E.g. to start a raw TCP input on port 5555, stop your container and recreate it, whilst appending `-p 5555:5555` to your run argument.
-Similarly, the same can be done for UDP by appending `-p 5555:5555/udp` option. Then you can send raw text to Graylog like
-`echo 'first log message' | nc localhost 5555`
+For example, to start a Raw/Plaintext TCP input on port 5555, stop your container and recreate it, whilst appending ``-p 5555:5555`` to your ``docker run`` command.
 
-Persist log data
-----------------
+Similarly, the same can be done for UDP by appending ``-p 5555:5555/udp``. After that you can send plaintext messages to Graylog using the following command::
 
-In order to make the log data and configuration of Graylog persistent, you can use external volumes to store all data. In case of a container restart simply re-use the existing data from the former instances.
-Create the configuration directory and copy the default files::
+  echo 'first log message' | nc localhost 5555
 
-  mkdir /graylog/config
-  cd /graylog/config
-  wget https://raw.githubusercontent.com/Graylog2/graylog2-images/2.3/docker/config/graylog.conf
-  wget https://raw.githubusercontent.com/Graylog2/graylog2-images/2.3/docker/config/log4j2.xml
 
-The `docker-compose.yml` file looks like this::
+.. _configuration:
+
+Configuration
+-------------
+
+Every configuration option can be set via `environment variables <https://github.com/Graylog2/graylog2-server/blob/2.3/misc/graylog.conf>`__.
+Simply prefix the parameter name with ``GRAYLOG_`` and put it all in upper case.
+
+For example, setting up the SMTP configuration for sending Graylog alert notifications via email, the ``docker-compose.yml`` would look like this::
 
   version: '2'
   services:
     mongo:
       image: "mongo:3"
+      # Other settings [...]
+    elasticsearch:
+      image: docker.elastic.co/elasticsearch/elasticsearch:5.5.1
+      # Other settings [...]
+    graylog:
+      image: graylog/graylog:2.3.0-1
+      # Other settings [...]
+      environment:
+        GRAYLOG_TRANSPORT_EMAIL_ENABLED: "true"
+        GRAYLOG_TRANSPORT_EMAIL_HOSTNAME: smtp
+        GRAYLOG_TRANSPORT_EMAIL_PORT: 25
+        GRAYLOG_TRANSPORT_EMAIL_USE_AUTH: "false"
+        GRAYLOG_TRANSPORT_EMAIL_USE_TLS: "false"
+        GRAYLOG_TRANSPORT_EMAIL_USE_SSL: "false"
+
+Another option would be to store the configuration file outside of the container and edit it directly (see :ref:`persisting-data`).
+
+
+.. _persisting-data:
+
+Persisting data
+---------------
+
```

Comment:
This section is a bit hard to read, maybe splitting it up into `Persisting data` and `External configuration files`?